packages/ui-react/src/components/LayoutGrid/LayoutGrid.tsx

```diff
@@ -0,0 +1,153 @@
+
+    event.preventDefault();
+
+    const isOnFirstColumn = currentColumnIndex === 0;
+    const isOnLastColumn = currentColumnIndex === columnCount - 1;
+    const isOnFirstRow = currentRowIndex === 0;
+    const isOnLastRow = currentRowIndex === rowCount - 1;
+    const maxRightLastRow = (data.length % columnCount || columnCount) - 1; // Never go beyond last item
+    const maxRight = isOnLastRow ? maxRightLastRow : columnCount - 1;
+
+    switch (key) {
+      case 'ArrowLeft':
+        if (isOnFirstColumn && !isOnFirstRow) {
+          // Move to last of previous row
+          setCurrentColumnIndex(columnCount - 1);
+          setCurrentRowIndex((current) => Math.max(current - 1, 0));
+
+          return;
+        }
+        return setCurrentColumnIndex((current) => Math.max(current - 1, 0));
+      case 'ArrowRight':
+        if (isOnLastColumn && !isOnLastRow) {
+          // Move to first of next row
+          setCurrentColumnIndex(0);
+          setCurrentRowIndex((current) => Math.min(current + 1, rowCount - 1));
+
+          return;
+        }
+        return setCurrentColumnIndex((current) => Math.min(current + 1, maxRight));
+      case 'ArrowUp':
+        return setCurrentRowIndex((current) => Math.max(current - 1, 0));
+      case 'ArrowDown':
+        return setCurrentRowIndex((current) => Math.min(current + 1, rowCount - 1));
+      case 'Home':
+        if (ctrlKey) {
+          setCurrentRowIndex(0);
+        }
+        return setCurrentColumnIndex(0);
+      case 'End':
+        if (ctrlKey) {
+          setCurrentRowIndex(maxRight);
+          setCurrentColumnIndex(maxRightLastRow);
+
+          return;
+        }
+        return setCurrentColumnIndex(rowCount - 1);
+      case 'PageUp':
+        return setCurrentRowIndex((current) => Math.max(current - 1, 0));
+      case 'PageDown':
+        return setCurrentRowIndex((current) => Math.min(current + 1, rowCount - 1));
+      default:
+        return;
+    }
+  });
+
+  const originalScrollBehavior = useRef<string | null>(null);
+
+  useEffect(() => {
+    if (focused) {
+      document.addEventListener('keydown', handleKeyDown);
+
+      // Prevent immediate page scrolling when out-of-viewport element gets focus
+      originalScrollBehavior.current = document.documentElement.style.scrollBehavior;
+      document.documentElement.style.scrollBehavior = 'smooth';
+    }
+
+    return () => {
+      document.documentElement.style.scrollBehavior = originalScrollBehavior.current || '';
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [focused, handleKeyDown, columnCount, rowCount]);
+
+  // Set DOM focus to a focusable element within the currently focusable grid cell
+  useLayoutEffect(() => {
+    if (!focused) return;
+
+    const gridCell = document.getElementById(`layout_grid_${currentRowIndex}-${currentColumnIndex}`) as HTMLDivElement | null;
```

Comment:
Although not a problem now, should we ensure the cell IDs are unique? If the GridLayout is rendered twice, it will conflict.